src/ucp/wireup/wireup.c

```diff
@@ -1060,14 +1061,41 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep,
         return;
     }
 
-    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
-        ucs_assert(ep->uct_eps[lane] != NULL);
-    }
-
     old_key = &ucp_ep_config(ep)->key;
 
     ucp_ep_config_lanes_intersect(old_key, new_key, reuse_lane_map);
 
+    if (ucp_ep_config_key_has_cm_lane(new_key)) {
+        for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+            ucs_assert(ep->uct_eps[lane] != NULL);
+        }
+
+        /* CM lane has to be re-used by the new EP configuration */
+        ucs_assert(reuse_lane_map[ucp_ep_get_cm_lane(ep)] != UCP_NULL_LANE);
+        /* wireup lane has to be selected for the old configuration */
+        ucs_assert(old_key->wireup_msg_lane != UCP_NULL_LANE);
+        /* wireup lane hasn't to be selected for the new configuration */
```

Comment:
`/* wireup lane hasn't been selected by the new configuration: only this function should select it */`